localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_request.py

```diff
@@ -66,12 +67,16 @@ def __call__(
         body, request_override = self.render_request_template_mapping(
             context=context, template=request_template
         )
+        # TODO: log every override that happens afterwards (in a loop on `request_override`)
         merge_recursive(request_override, request_data_mapping, overwrite=True)
 
-        # TODO: extract the code under into its own method
-
-        # TODO: create helper method to render the different URI with stageVariables and parameters
-        rendered_integration_uri = integration["uri"]  # use request_data_mapping["path"]
+        # looks like the stageVariables rendering part is done in the Integration part in AWS
+        # but we can avoid duplication by doing it here for now
+        rendered_integration_uri = self.render_integration_uri(
+            uri=integration["uri"],
+            path_parameters=request_data_mapping["path"],
+            stage_variables=context.stage_variables,
+        )
```

Comment:
I guess we are ignoring the cross-account thing for now! :wink:

Reply:
Added for a comment/TODO for it 😄